src/marker.rs

```diff
@@ -478,6 +478,8 @@ impl<'py> Python<'py> {
                 gil::GIL_COUNT.with(|c| c.set(self.count));
                 unsafe {
                     ffi::PyEval_RestoreThread(self.tstate);
+                    // Update counts of PyObjects / Py that were cloned or dropped during `f`.
+                    gil::POOL.update_counts(Python::assume_gil_acquired());
```

Comment:
I wonder whether we should package up `RestoreGuard` in `mod gil` to centralize the usage of `Save/RestoreThread`, accessing `GIL_COUNT` and calling `update_counts` in that module. (I suspect that having this in separate modules, i.e. making `GIL_COUNT` a `pub(crate)` item, might have been the root cause of this slipping in.)